DataGrid demo: Rewrite Toolbar Customization (#77)
DevExpress/devextreme-demos: JSDemos/Demos/DataGrid/ToolbarCustomization/description.md

> The toolbar contains predefined and custom controls that execute various commands.

Reviewer: This sentence wouldn't be needed if you accept my correction to the second paragraph. They are almost saying the same thing.

Author: Will remove it.

> Toolbar provides additional control elements. To add and customize the toolbar, implement the onToolbarPreparing option. You can configure an existent element or add a DevExtreme or custom widget in this function. For this, manipulate the toolbarOptions.items array located in the parameter object.

Reviewer 2: ```suggestion
The toolbar contains predefined and custom controls that execute various commands.

The **DataGrid** uses the DevExtreme [Toolbar](/Demos/WidgetsGallery/Demo/Toolbar/Overview/) component as the toolbar. To customize the toolbar, implement the [onToolbarPreparing](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onToolbarPreparing) function. Its parameter contains the [Toolbar's configuration](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/) in the **toolbarOptions** field. You can access toolbar items via the **toolbarOptions**.[items](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/) array. Modify this array to add or remove toolbar items.
```

Author: It is clear that a toolbar contains controls and that the DE Toolbar widget implements the toolbar functionality. Keeping this in mid, I used a single paragraph instead of two, made some changes and used the suggested options. '*Additional* controls' is used to highlight that the widget works properly without them. The paragraph is as follows:

The Toolbar widget contains additional controls for DevExtreme widgets. To add and customize DataGrid's Toolbar, implement the onToolbarPreparing function. Its parameter's toolbarOptions field contains the [configuration]() object. Modify the the **toolbarOptions**.[items] array to manipulate and add elements to the Toolbar. They can be DevExtreme widgets and custom elements.

> The **DataGrid** uses the DevExtreme [Toolbar](/Demos/WidgetsGallery/Demo/Toolbar/Overview/) component as the toolbar. To customize the toolbar, implement the [onToolbarPreparing](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onToolbarPreparing) function. Its parameter contains the [Toolbar's configuration](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/) in the **toolbarOptions** field. You can access toolbar items via the **toolbarOptions**.[items](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/) array. Modify this array to add or remove toolbar items.

Reviewer: I'm mostly trying to avoid the repetition of the word 'Toolbar' that appears in almost every sentence, twice in some sentences. 

```suggestion
The **DataGrid** uses an integrated DevExtreme [Toolbar](/Demos/WidgetsGallery/Demo/Toolbar/Overview/) component to display built-in and custom commands. To add or remove toolbar items, handle the [onToolbarPreparing](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onToolbarPreparing) event. Use the parameter's **toolbarOptions** field to access the [Toolbar's configuration](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/) object. This object includes the command collection: the **toolbarOptions**.[items](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/) array. 
```

Author: ```suggestion
The **DataGrid** uses an integrated DevExtreme [Toolbar](/Demos/WidgetsGallery/Demo/Toolbar/Overview/) component to display predefined and custom commands. To add or remove toolbar items, implement the [onToolbarPreparing](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onToolbarPreparing) function. Use its parameter's **toolbarOptions** field to access the [Toolbar's configuration](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/) object. This object includes the command collection: the **toolbarOptions**.[items](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/) array.
```

1) 'predefined' because it stands for DevExtreme widgets and Predefined controls (e.g. columnChooser).

2) 'implement the onToolbarPreparing function' because onToolbarPreparing is a function.

Is it good?

Reviewer: Yes, this is fine, thanks.

* Button widget to collapse/expand data groups; it uses the **option** method of DataGrid API to set value for the grouping.autoExpandAll option

* Button widget to refresh grid data; this element uses the **refresh** method of DataGrid API

* SelectBox to specify whether to group by State ot Employee

> * Custom HTML element to display the number of data groups.

Reviewer 2: ```suggestion
This demo illustrates how to add the following elements to the toolbar:

- **DevExtreme widgets**        
Specify the [widget](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#widget) that you want to add and its [options](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#options). In this demo, we added the [Button](/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/) and [SelectBox](/Demos/WidgetsGallery/Demo/SelectBox/Overview/) widgets.

- **Custom elements**     
Specify a [template](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#template) for your custom element. In this demo, the custom element displays the total group count.
```

Author: I used the 'DevExtreme widgets' and 'Custom elements' headers, but modified the paragraphs:

**DevExtreme widgets**
To add a DevExtreme widget to the toolbar, specify the [widget] and [options] fields of the object in the **toolbarOptions**.[items] array.

In this demo, the following widgets are added:

* Button widget to collapse/expand data groups;

* Button widget to refresh grid data;

* SelectBox to specify whether to group by State ot Employee.

**Custom elements**
To add a custom element, specify the template field of the object in **toolbarOptions**.[items] array. In this demo, the counter to display the number of data groups is added.

> The toolbar also consists of the Column Chooser button. To enable this element, add the columnChooser object to the DataGrid and set its enabled option to true.

Reviewer 2: ```suggestion
Toolbar items in this demo also include the Column Chooser button. This is a predefined button that the **DataGrid** added because the [columnChooser](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columnChooser/) is enabled.
```

Author: Shortend it to 'This demo Toolbar also includes the Column Chooser button. The **DataGrid** added it because the [columnChooser]() is enabled.'

> Specify the [widget](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#widget) that you want to add and its [options](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#options). In this demo, we added the [Button](/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/) and [SelectBox](/Demos/WidgetsGallery/Demo/SelectBox/Overview/) widgets.

Reviewer: The way you write the first sentence looks like they need to specify widget's settings. "Specify the widget and its options". But I guess you specify the widget, together with its own settings. And you also specify the options for the toolbar item that contains the widget. 

In both sentences I'd use indefinite articles. 

```suggestion
Specify a [widget](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#widget) that you want to display in a toolbar item and the item's [options](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#options). In this demo, we extended the grid's command collection with a [Button](/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/) and a [SelectBox](/Demos/WidgetsGallery/Demo/SelectBox/Overview/).
```

Author: A developer has to specify the widget name in the 'widget' option and this widget's options in the 'options' option. That's why the paragrpah is as follows:
```suggestion
Specify a [widget](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#widget) that you want to add and its [options](/Documentation/ApiReference/UI_Widgets/dxToolbar/Configuration/items/#options). In this demo, we extended the toolbar's item collection with a [Button](/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/) and a [SelectBox](/Demos/WidgetsGallery/Demo/SelectBox/Overview/).
```

Can it be applied?

Reviewer: Yes, this version is good. I did some research - apparently I misunderstood what "options" were. Sorry.

> Toolbar items in this demo also include the Column Chooser button. This is a predefined button that the **DataGrid** added because the [columnChooser](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columnChooser/) is enabled.

Reviewer: If other section have titles, I guess we should add a title for this paragraph - something like "Built-in commands".

Author: ```suggestion
- **Predefined controls**        
The availability of predefined controls depends on whether a specific **DataGrid** feature is enabled. For example, this demo enables the [columnChooser](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columnChooser/), and the toolbar displays the Column Chooser button. You can customize the predefined controls in the **onToolbarPreparing** function. Refer to its [description](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onToolbarPreparing) for a code example.
```

?

Reviewer: Looks good to me. Thanks
